hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlCreateSnapshot.java

```diff
@@ -45,10 +46,15 @@
     public SqlCreateSnapshot(
             SqlIdentifier snapshotName,
             SqlIdentifier jobName,
+            boolean replace,
             SqlParserPos pos
     ) {
         super(OPERATOR, pos, true, false);
 
+        if (!replace) {
+            throw QueryException.error("The OR REPLACE option is required for CREATE SNAPSHOT");
```

Comment:
Throw `SqlValidatorException` (see `ParserResource`)?

Reply:
I don't like the indirection much, I don't see a benefit to it. It technically isn't a validator exception, but parsing exc: the parser fails to create a SqlNode.
We do this style already, for example here: https://github.com/hazelcast/hazelcast-jet/blob/master/hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlCreateJob.java#L117
The user will receive `HazelcastSqlException` with `code=PARSING` anyway.

Reply:
Exactly, if it's a parsing one maybe we should throw a _parsing_ exception, giving user position etc., `QueryException` does not seem to be fitted to that.

Reply:
Got it, fixed. But the exception marks the whole command...